openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionSignatureGenerator.java

```diff
@@ -370,46 +367,76 @@ public Node getPathParameters(Parameter parameter, NodeList<AnnotationNode> para
+            throws BallerinaOpenApiException {
+        boolean isRequired = false;
+        Schema schema = parameter.getValue().getSchema();
+        IdentifierToken paramName = createIdentifierToken(getValidName(parameter.getKey(), false));
+        List<AnnotationNode> parameterAnnotationNodeList = new ArrayList<>();
+
+        if (parameter.getValue().getDescription() != null) {
+            MarkdownParameterDocumentationLineNode paramAPIDoc = DocCommentsGenerator.createAPIParamDoc(
+                    getValidName(parameter.getKey(), false), parameter.getValue().getDescription());
+            remoteFunctionDoc.add(paramAPIDoc);
         }
-        if (parameter.getRequired()) {
+        DocCommentsGenerator.extractDisplayAnnotation(parameter.getValue().getExtensions(),
+                parameterAnnotationNodeList);
+
+        if (parameter.getValue().getRequired() != null) {
+            isRequired = parameter.getValue().getRequired();
+        }
+        return getHeader(isRequired, schema, paramName, createNodeList(parameterAnnotationNodeList));
+    }
+
+    /**
+     * Create header for header parameter and encoding.
+     */
+    private Node getHeader(boolean isRequiredHeader, Schema schema, IdentifierToken paramName,
+                           NodeList<AnnotationNode> parameterAnnotationNodeList) throws BallerinaOpenApiException {
+        if (isRequiredHeader) {
+            String type = convertOpenAPITypeToBallerina(schema.getType().trim());
+            if (schema instanceof ArraySchema) {
+                ArraySchema arraySchema = (ArraySchema) schema;
+                if (arraySchema.getItems().get$ref() != null) {
+                    type = extractReferenceType(arraySchema.getItems().get$ref()) + SQUARE_BRACKETS;
+                } else {
+                    type = convertOpenAPITypeToBallerina(arraySchema.getItems().getType().trim()) + SQUARE_BRACKETS;
+                }
+            }
             BuiltinSimpleNameReferenceNode typeName = createBuiltinSimpleNameReferenceNode(null,
                     createIdentifierToken(type));
-            IdentifierToken paramName = createIdentifierToken(getValidName(parameter.getName().trim(),
-                    false));
             return createRequiredParameterNode(parameterAnnotationNodeList, typeName, paramName);
         } else {
-            IdentifierToken paramName = createIdentifierToken(getValidName(parameter.getName().trim(),
-                    false));
             if (schema.getDefault() != null) {
                 BuiltinSimpleNameReferenceNode typeName = createBuiltinSimpleNameReferenceNode(null,
-                        createIdentifierToken(convertOpenAPITypeToBallerina(parameter.getSchema().getType().trim())));
+                        createIdentifierToken(convertOpenAPITypeToBallerina(schema.getType().trim())));
                 LiteralValueToken literalValueToken;
                 if (schema.getType().equals("string")) {
```

Comment:
```suggestion
                if (schema.getType().equals(STRING)) {
```

Reply:
Resolved in https://github.com/ballerina-platform/openapi-tools/commit/ed82b26fe32a2c110366b7a338e378c554829742